aloha-core/src/test/scala/com/eharmony/aloha/models/exploration/EpsilonGreedyModelTest.scala

```diff
@@ -0,0 +1,75 @@
+package com.eharmony.aloha.models.exploration
+
+import com.eharmony.aloha.ModelSerializationTestHelper
+import com.eharmony.aloha.factory.ModelFactory
+import com.eharmony.aloha.id.ModelId
+import com.eharmony.aloha.models.{AnySemanticsWithoutFunctionCreation, CloserTesterModel, ConstantModel}
+import com.eharmony.aloha.score.conversions.ScoreConverter.Implicits._
+import com.eharmony.aloha.semantics.func.GenFunc0
+import org.junit.Assert._
+import org.junit.Test
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+
+import scala.collection.{immutable => sci}
+
+/**
+  * Created by jmorra on 2/26/16.
+  */
+class EpsilonGreedyModelTest extends ModelSerializationTestHelper {
+  private[this] val reader = EpsilonGreedyModel.Parser.modelJsonReader[Any, String](ModelFactory(ConstantModel.parser), Option(AnySemanticsWithoutFunctionCreation))
+  private[this] val delta = 0.00001f
+  implicit val audit = true
+
+  @Test def testSerialization() {
+    val constantPolicy = ConstantModel(Right(1), ModelId(2, "abc"))
+    val m = EpsilonGreedyModel(ModelId(3, "def"), constantPolicy, 0.1f, null, sci.IndexedSeq(1, 2, 3))
+    val m1 = serializeDeserializeRoundTrip(m)
+    assertEquals(m, m1)
+
+    val m2 = EpsilonGreedyModel(ModelId(3, "def"), constantPolicy, 0.1f, null, sci.IndexedSeq("1", "2", "3"))
+    val m3 = serializeDeserializeRoundTrip(m2)
+    assertEquals(m2, m3)
+  }
+
+  @Test def testClosed() {
+    val sub = new CloserTesterModel[Int]()
+    EpsilonGreedyModel(ModelId.empty, sub, 0.1f, GenFunc0("", (s: String) => 1l), sci.IndexedSeq(1, 2)).close()
+    assertTrue(sub.isClosed)
+  }
+
+  @Test def random() {
+    val epsilon = 0.9f
+    val m = makeModel(1, epsilon, 1)
+    val s = m.getScore(null)
+    assertEquals(s._2.get.getScore.getProbability, epsilon / 3, delta)
+    assertEquals(s._1.right.get, "b")
+  }
+
+  @Test def policy() {
+    val epsilon = 0.1f
```

Comment:
epsilon should equal 0.  And instead of directly checking that 

```scala
s._1.right.get, "a"
```

you should create the RNG with the seed and check that `m.classLabels(RNG return value - 1)` equals `s._1.right.get`.  This way if you update MWT and they change the code, it might fail.  This would be good because then at least you would realize that something changed and that the model worked differently.